esp-hal/src/gpio/interconnect.rs

```diff
@@ -0,0 +1,384 @@
+//! Peripheral signal interconnect using IOMUX or GPIOMUX.
+
+use crate::{
+    gpio::{
+        self,
+        AlternateFunction,
+        AnyPin,
+        DummyPin,
+        GpioPin,
+        GpioProperties,
+        InputPin,
+        Level,
+        OutputSignalType,
+        PeripheralInputPin,
+        PeripheralOutputPin,
+        Pin,
+        FUNC_IN_SEL_OFFSET,
+        GPIO_FUNCTION,
+        INPUT_SIGNAL_MAX,
+        OUTPUT_SIGNAL_MAX,
+    },
+    peripheral::Peripheral,
+    peripherals::GPIO,
+    private::{self, Sealed},
+};
+
+/// Multiple input signal can be connected to one pin.
+pub struct InputSignal {
+    pin: AnyPin,
+    is_inverted: bool,
+}
+
+impl Clone for InputSignal {
+    fn clone(&self) -> Self {
+        Self {
+            pin: unsafe { self.pin.clone_unchecked() },
+            is_inverted: self.is_inverted,
+        }
+    }
+}
+
+impl Peripheral for InputSignal {
+    type P = Self;
+
+    unsafe fn clone_unchecked(&mut self) -> Self::P {
+        self.clone()
+    }
+}
+
+impl Sealed for InputSignal {}
+
+impl InputSignal {
+    pub(crate) fn new(pin: AnyPin) -> Self {
+        Self {
+            pin,
+            is_inverted: false,
+        }
+    }
+
+    /// Inverts the peripheral's input signal.
+    pub fn invert(&mut self) {
+        self.is_inverted = !self.is_inverted;
+    }
+
+    /// Inverts the peripheral's input signal.
```

Comment:
comment is for the setter

Reply:
I mean this is a setter, too, just with move semantics that is IMO better suited for the simple "grab signal-invert-pass to peripheral" case. I'm not sure how to word the docs, or if we want to keep both variants of the setter, so I'm open to suggestions.

Reply:
ah - sure - I need more coffee probably